### DISCUSSION TITLE
Exception occurs when calling the value() method using json_pointer

### DISCUSSION THREAD

**Discussion by George-J-Cho** [nlohmann/json#4470] [Q&A]
Exceptions occur in certain situations in the following code lines.
```
int Range = jsonResult.value(nlohmann::json::json_pointer("/act/cpC/sc/0/parameters/Range"), 65535);
```

I am a developer of C++ Junior.

As far as I'm debunking
If there is no path value corresponding to the jsonResult, it works fine
It works fine when there is a path value,

Exception occurs when the middle part of the path is transferred to the array.

Is it the intended result? Is it an unknown bug?
I expected that even if part of the path was wrong, it would be the default value of 65535.

If anyone knows, please help me.
Thank you.

**Comment by gregmarr** [comment]
The `value()` function will throw an exception if you provide it with a json pointer that doesn't exist.

**Reply by George-J-Cho** [reply]
Thank you very much for your reply.

If the Json object does not exist, the default value is not used.
There are many cases like this in projects.

Would it be difficult to improve value() without adding exception handling?

**Reply by gregmarr** [reply]
It would be a breaking behavior change, and @nlohmann indicated in another PR recently that he didn't want to change that behavior without a major release.

@nlohmann It has been almost 8 years since 3.0.  I thought we had a v4 planning document somewhere but I can't find it right now.

**Reply by nlohmann** [reply]
@gregmarr I don't have a specific plan. I am honestly a bit afraid to have this one shot at a new major version, because it feels like a lot of pressure to get this right.

**Reply by gregmarr** [reply]
I was mostly looking for "here's a list of things that we want to fix in a new major version".  Maybe just a `v4.0` tag would work for now.

**Reply by nlohmann** [reply]
Yes, I can do that.

**Reply by nlohmann** [reply]
@gregmarr I created a label "V4 candidate" that can be used as a quick bookmark for these issues. But we should have a proper roadmap as well.

**Reply by George-J-Cho** [reply]
@nlohmann 
If there is a problem with the path, will it be improved to be specified as the default value? (In the v4 version)
I'm sorry for the awkward expression because I'm not good at English.
Thank you so much for your answer.
